djangobaseview/guestbook/models.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from google.appengine.api import users
+from google.appengine.ext import ndb
+
+DEFAULT_GUESTBOOK_NAME = 'default_guestbook'
+
+
+# We set a parent key on the 'Greetings' to ensure that they are all in the same
+# entity group. Queries across the single entity group will be consistent.
+# However, the write rate should be limited to ~1/second.
+
+def guestbook_key(guestbook_name=DEFAULT_GUESTBOOK_NAME):
+    '''Constructs a Datastore key for a Guestbook entity with guestbook_name.'''
+    return ndb.Key('Guestbook', guestbook_name)
+
+
+class Greeting(ndb.Model):
+    '''Models an individual Guestbook entry.'''
+    author = ndb.UserProperty()
```

Comment:
hiện tại toàn bộ code đang dùng 4 space, 

hãy chỉnh pycharm lại để tab là tab, không phải 4 space